tests/models/hydrology/test_hydrology_tools.py

```diff
@@ -31,80 +16,107 @@ def test_setup_hydrology_input_current_timestep(
 
     # check if climate values are selected correctly
     np.testing.assert_allclose(
         np.sum(result["current_precipitation"], axis=1),
         (dummy_climate_data["precipitation"].isel(time_index=0)).to_numpy(),
     )
     np.testing.assert_allclose(
-        result["subcanopy_temperature"], dummy_climate_data["air_temperature"][11]
+        result["surface_temperature"],
+        dummy_climate_data["air_temperature"]
+        .isel(layers=fixture_core_components.layer_structure.role_indices["surface"])
+        .squeeze(),
     )
     np.testing.assert_allclose(
-        result["subcanopy_humidity"], dummy_climate_data["relative_humidity"][11]
+        result["surface_humidity"],
+        dummy_climate_data["relative_humidity"]
+        .isel(layers=fixture_core_components.layer_structure.role_indices["surface"])
+        .squeeze(),
     )
+    # The reference data is a time series with cell id in axis 0, the result has cell_id
+    # on axis 1, so need to extract as 2d and rotate to match.
     np.testing.assert_allclose(
-        result["subcanopy_pressure"],
-        (dummy_climate_data["atmospheric_pressure_ref"].isel(time_index=0)).to_numpy(),
+        result["surface_pressure"],
+        dummy_climate_data["atmospheric_pressure_ref"]
+        .isel(time_index=[0])
+        .squeeze()
+        .to_numpy()
+        .T,
     )
     np.testing.assert_allclose(
-        result["soil_moisture_mm"], DataArray(np.full((2, 3), 50.0))
+        result["soil_moisture_mm"],
+        DataArray(np.tile([[5], [500]], fixture_core_components.grid.n_cells)),
     )
 
 
-def test_initialise_soil_moisture_mm(fixture_core_components):
+@pytest.mark.parametrize(
+    argnames="init_soilm, expected",
+    argvalues=(
+        pytest.param(0.5, np.array([[250], [250]]), id="scalar_init_soilm"),
+        # pytest.param(0.5, np.array([[125], [375]]), id="scalar_init_soilm"),
+    ),
+)
+def test_initialise_soil_moisture_mm(fixture_core_components, init_soilm, expected):
     """Test soil moisture is initialised correctly."""
 
     from virtual_ecosystem.models.hydrology.hydrology_tools import (
         initialise_soil_moisture_mm,
     )
 
+    layer_structure = fixture_core_components.layer_structure
+
     result = initialise_soil_moisture_mm(
-        soil_layer_thickness=np.array([[10, 10, 10, 10], [100, 100, 100, 100]]),
-        layer_structure=fixture_core_components.layer_structure,
-        n_cells=4,
-        initial_soil_moisture=0.5,
+        layer_structure=layer_structure,
+        soil_layer_thickness=np.tile(
+            layer_structure.soil_layer_thickness[:, None] * 1000,
+            fixture_core_components.grid.n_cells,
+        ),
+        initial_soil_moisture=init_soilm,
+    )
+    # The fixture is configured with soil layers [-0.25, -1.0]
+    exp_result = DataArray(np.broadcast_to(expected, (2, 4)))
+    np.testing.assert_allclose(
+        result[layer_structure.role_indices["all_soil"]], exp_result
     )
```

Comment:
This is not matching up the dimensions correctly, the values should differ. Why can we not do `soil_layer_thickness * 1000`?

Reply:
I don't think we can - this is checking the soil moisture.